### PR TITLE
Migrate ngraph backend test of arithmetic2 operations

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - add sin/cos/asin/sinh/cosh/tanh/asinh in template plugin reference test
 - add atanh activation type in CPU/GPU SLT
 - check SLT parameters for sin/cos/tan/asin/sinh/cosh/tanh/asinh/acosh/atanh
 - add tanh in ngraph type_prop test
 - add sin/asin in ngraph visitor API test
 - remove sin/cos/asin/sinh/cosh/tanh/asinh in ngraph backend test
 - Remove boolean type from sin/cos/tan/asin/sinh/cosh_evaluate

### Tickets:
 - 63941
 - 63943
 - 63950
 - 63951
 - 63991
 - 63992
 - 63997
 - 63998
 - 64057
 - 64058

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/activation.cpp [hunk 1]
```diff
@@ -42,13 +42,13 @@ const std::map<ActivationTypes, std::vector<std::vector<float>>> activationTypes
...
         {Asinh,                 {}},
         {Atan,                  {}},
+        {Atanh,                  {}},
         {Cos,                   {}},
         {Cosh,                  {}},
         {Floor,                 {}},
         {Sin,                   {}},
         {Sinh,                  {}},
         {Sqrt,                  {}},
-        {Tan,                   {}},
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/activation.cpp:51]: why did you remove `Tan` op?

Comment (wilson-seok) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/activation.cpp:51]: It is duplicated in line#31.

File: docs/template_plugin/tests/functional/op_reference/asinh.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    Exec();
+}
+
+}  // namespace
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_Asinh_With_Hardcoded_Refs, ReferenceAsinhLayerTest,
+    ::testing::Values(
+        Builder {}
+            .input({{11}, element::f16, std::vector<ngraph::float16> {0.f, 1.f, -1.f, 2.f, -2.f, 3.f, -3.f, 4.f, 5.f, 10.f, 100.f}})
```
Comment (iefode) [inline docs/template_plugin/tests/functional/op_reference/asinh.cpp:64]: I think, it will better to use such [mechanism](https://github.com/openvinotoolkit/openvino/blob/243028a40871dd9b2ed95f92c75063ebd1b140c1/docs/template_plugin/tests/functional/op_reference/gather_tree.cpp#L66). I mean to use templates to combine different datatypes for vectors

Comment (wilson-seok) [reply docs/template_plugin/tests/functional/op_reference/asinh.cpp:64]: Sometimes I need to modify each element of expected values when int ot uint case. I mean that the value from backend test used -2 and 2 for input value -3 and 3 in data type int32/64. But type cast returns -1 and 1 from -1.81844646f and 1.81844646f and they causes test failure. So I think that template type is not proper for this case.

**Review by ilyachur — APPROVED** [review]

**Comment by wilson-seok** [comment]
@ilyachur Could you please merge this PR if you have the permission?

**Comment by ilyachur** [comment]
I would like to wait review from @iefode

**Review by iefode — APPROVED** [review]
